no field `hsi48`, `pll1_p` on type Temp when building `examples/stm32f0` for `stm32f072c8`

To reproduce:

`cd examples/stm32f0`
Change `stm32f091rc` to `stm32f072c8` (keep rest of `Cargo.toml` the same)
`cargo build`

The result is:

```
error[E0609]: no field `hsi48` on type `Temp`
   --> /tmp/blorg/target/thumbv6m-none-eabi/debug/build/embassy-stm32-fc57c2643893afc2/out/_macros.rs:1:347
    |
1   |   ...rcc :: Clocks { hclk1 : all . hclk1 , hsi : all . hsi , hsi48 : all . hsi48 , lse : all . lse , pclk1 : all . pclk1 , pclk1_tim : all . pclk1_tim , pclk2 : all...
    |                                                                            ^^^^^ unknown field
    |
   ::: /home/ablyve/tmp/embassy/embassy-stm32/src/rcc/f0.rs:162:5
    |
162 | /     set_clocks!(
163 | |         hsi: None,
164 | |         lse: None,
165 | |         sys: Some(Hertz(real_sysclk)),
...   |
171 | |         rtc: rtc,
172 | |     );
    | |_____- in this macro invocation
    |
    = note: this error originates in the macro `set_clocks` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0609]: no field `pll1_p` on type `Temp`
   --> /tmp/blorg/target/thumbv6m-none-eabi/debug/build/embassy-stm32-fc57c2643893afc2/out/_macros.rs:1:492
    |
1   |   ..., pclk2 : all . pclk2 , pclk2_tim : all . pclk2_tim , pll1_p : all . pll1_p , rtc : all . rtc , sys : all . sys , }) ; } } ; }#[allow(unused)]
    |                                                                           ^^^^^^ unknown field
    |
   ::: /home/ablyve/tmp/embassy/embassy-stm32/src/rcc/f0.rs:162:5
    |
162 | /     set_clocks!(
163 | |         hsi: None,
164 | |         lse: None,
165 | |         sys: Some(Hertz(real_sysclk)),
...   |
171 | |         rtc: rtc,
172 | |     );
    | |_____- in this macro invocation
    |
    = note: this error originates in the macro `set_clocks` (in Nightly builds, run with -Z macro-backtrace for more info)
```

More info:

```
rustc --version                                                                                                                                                  (main !?)
rustc 1.75.0 (82e1608df 2023-12-21)
```

same issue on stm32l431cb for the field pllsai2_p
ie.
```
error[E0609]: no field `pllsai2_p` on type `Temp`
   --> embassy/examples/stm32l4/target/thumbv7em-none-eabi/release/build/embassy-stm32-aab123763c6d6b70/out/_macros.rs:1:595
    |
1   |   ...l1_q , pllsai1_p : all . pllsai1_p , pllsai2_p : all . pllsai2_p , rtc : all . rtc , sai1_extclk : all . sai1_extclk , sys : all . sys...
    |                                                             ^^^^^^^^^ unknown field
    |
   ::: /home/ragarnoy/CLionProjects/embassy/embassy-stm32/src/rcc/l.rs:378:5
    |
378 | /     set_clocks!(
379 | |         sys: Some(sys_clk),
380 | |         hclk1: Some(hclk1),
381 | |         #[cfg(any(stm32l4, stm32l5, stm32wb, stm32wl))]
...   |
423 | |         lse: None,
424 | |     );
    | |_____- in this macro invocation
    |
    = note: this error originates in the macro `set_clocks` (in Nightly builds, run with -Z macro-backtrace for more info)

```

--- a/ci.sh
+++ b/ci.sh
@@ -84,6 +84,7 @@ cargo batch  \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,defmt,time \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,defmt,exti \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,defmt \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features stm32f072c8,defmt,exti,time-driver-any,time \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features stm32f401ve,defmt,exti,time-driver-any \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features stm32f405zg,defmt,exti,time-driver-any \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features stm32f407zg,defmt,exti,time-driver-any \
@@ -110,6 +111,7 @@ cargo batch  \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features stm32h755zi-cm7,defmt,exti,time-driver-any,time \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features stm32h725re,defmt,exti,time-driver-any,time \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features stm32h7b3ai,defmt,exti,time-driver-any,time \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features stm32l431cb,defmt,exti,time-driver-any,time \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features stm32l476vg,defmt,exti,time-driver-any,time \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features stm32l422cb,defmt,exti,time-driver-any,time \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features stm32wb15cc,defmt,exti,time-driver-any,time \

--- a/embassy-stm32/src/rcc/f0.rs
+++ b/embassy-stm32/src/rcc/f0.rs
@@ -11,14 +11,13 @@ pub const HSI_FREQ: Hertz = Hertz(8_000_000);
 ///
 /// hse takes precedence over hsi48 if both are enabled
 #[non_exhaustive]
-#[derive(Default)]
 pub struct Config {
     pub hse: Option<Hertz>,
     pub bypass_hse: bool,
     pub usb_pll: bool,
 
-    #[cfg(not(stm32f0x0))]
-    pub hsi48: bool,
+    #[cfg(crs)]
+    pub hsi48: Option<super::Hsi48Config>,
 
     pub sys_ck: Option<Hertz>,
     pub hclk: Option<Hertz>,
@@ -27,12 +26,31 @@ pub struct Config {
     pub ls: super::LsConfig,
 }
 
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            hse: Default::default(),
+            bypass_hse: Default::default(),
+            usb_pll: Default::default(),
+            hsi48: Some(Default::default()),
+            sys_ck: Default::default(),
+            hclk: Default::default(),
+            pclk: Default::default(),
+            ls: Default::default(),
+        }
+    }
+}
+
 pub(crate) unsafe fn init(config: Config) {
     let sysclk = config.sys_ck.map(|v| v.0).unwrap_or(HSI_FREQ.0);
 
+    #[cfg(crs)]
+    let hsi48 = config.hsi48.map(|config| super::init_hsi48(config));
+    #[cfg(not(crs))]
+    let hsi48: Option<Hertz> = None;
+
     let (src_clk, use_hsi48) = config.hse.map(|v| (v.0, false)).unwrap_or_else(|| {
-        #[cfg(not(stm32f0x0))]
-        if config.hsi48 {
+        if hsi48.is_some() {
             return (48_000_000, true);
         }
         (HSI_FREQ.0, false)
@@ -169,5 +187,9 @@ pub(crate) unsafe fn init(config: Config) {
         pclk2_tim: Some(Hertz(pclk * timer_mul)),
         hclk1: Some(Hertz(hclk)),
         rtc: rtc,
+        hsi48: hsi48,
+
+        // TODO:
+        pll1_p: None,
     );
 }

--- a/embassy-stm32/src/rcc/l.rs
+++ b/embassy-stm32/src/rcc/l.rs
@@ -215,12 +215,9 @@ pub(crate) unsafe fn init(config: Config) {
     });
 
     #[cfg(crs)]
-    let _hsi48 = config.hsi48.map(|config| {
-        //
-        super::init_hsi48(config)
-    });
+    let hsi48 = config.hsi48.map(|config| super::init_hsi48(config));
     #[cfg(not(crs))]
-    let _hsi48: Option<Hertz> = None;
+    let hsi48: Option<Hertz> = None;
 
     let _plls = [
         &config.pll,
@@ -274,12 +271,12 @@ pub(crate) unsafe fn init(config: Config) {
     RCC.ccipr().modify(|w| w.set_clk48sel(config.clk48_src));
     #[cfg(any(rcc_l0_v2))]
     let clk48 = match config.clk48_src {
-        Clk48Src::HSI48 => _hsi48,
+        Clk48Src::HSI48 => hsi48,
         Clk48Src::PLL1_VCO_DIV_2 => pll.clk48,
     };
     #[cfg(any(stm32l4, stm32l5, stm32wb))]
     let clk48 = match config.clk48_src {
-        Clk48Src::HSI48 => _hsi48,
+        Clk48Src::HSI48 => hsi48,
         Clk48Src::MSI => msi,
         Clk48Src::PLLSAI1_Q => pllsai1.q,
         Clk48Src::PLL1_Q => pll.q,
@@ -393,6 +390,7 @@ pub(crate) unsafe fn init(config: Config) {
         msi: msi,
         #[cfg(any(rcc_l0_v2, stm32l4, stm32l5, stm32wb))]
         clk48: clk48,
+        hsi48: hsi48,
 
         #[cfg(not(any(stm32l0, stm32l1)))]
         pll1_p: pll.p,
@@ -406,6 +404,13 @@ pub(crate) unsafe fn init(config: Config) {
         pllsai1_q: pllsai1.q,
         #[cfg(any(stm32l4, stm32l5, stm32wb))]
         pllsai1_r: pllsai1.r,
+
+        #[cfg(not(any(stm32l47x, stm32l48x, stm32l49x, stm32l4ax, rcc_l4plus, stm32l5)))]
+        pllsai2_p: None,
+        #[cfg(not(any(stm32l47x, stm32l48x, stm32l49x, stm32l4ax, rcc_l4plus, stm32l5)))]
+        pllsai2_q: None,
+        #[cfg(not(any(stm32l47x, stm32l48x, stm32l49x, stm32l4ax, rcc_l4plus, stm32l5)))]
+        pllsai2_r: None,
 
         #[cfg(any(stm32l47x, stm32l48x, stm32l49x, stm32l4ax, rcc_l4plus, stm32l5))]
         pllsai2_p: pllsai2.p,

